### PR TITLE
[ENH] Add random_state parameter to BaseDistribution.sample for reproducible sampling

### PR DESCRIPTION
**Pull request by moksha-hub**
## Summary
Adds a `random_state` parameter to `BaseDistribution.sample()` and the private [_sample()](cci:1://file:///c:/Users/moksh/OneDrive/Desktop/Esoc/skpro/skpro/distributions/base/_base.py:1637:4-1661:76) so sampling is reproducible, as requested in #661.

## Changes
- [sample(n_samples=None)](cci:1://file:///c:/Users/moksh/OneDrive/Desktop/Esoc/skpro/skpro/distributions/base/_base.py:1604:4-1635:75) → [sample(n_samples=None, random_state=None)](cci:1://file:///c:/Users/moksh/OneDrive/Desktop/Esoc/skpro/skpro/distributions/base/_base.py:1604:4-1635:75)
- [_sample(n_samples=None)](cci:1://file:///c:/Users/moksh/OneDrive/Desktop/Esoc/skpro/skpro/distributions/base/_base.py:1637:4-1661:76) → [_sample(n_samples=None, random_state=None)](cci:1://file:///c:/Users/moksh/OneDrive/Desktop/Esoc/skpro/skpro/distributions/base/_base.py:1637:4-1661:76)
- Uses `sklearn.utils.check_random_state(random_state)` internally
- Backward compatible — calling [sample()](cci:1://file:///c:/Users/moksh/OneDrive/Desktop/Esoc/skpro/skpro/distributions/base/_base.py:1604:4-1635:75) with no args unchanged

## Usage
```python
d = Normal(mu=0, sigma=1)
s1 = d.sample(100, random_state=42)
s2 = d.sample(100, random_state=42)
# s1 == s2 — reproducible!

### CODE REVIEW

**Review by arnavk23 — COMMENTED** [review]
this changes the subclass interface. `sample()` now calls `_sample(..., random_state=...)`, which will **break any existing distribution subclass** that still defines `_sample(self, n_samples=None)` (TypeError for unexpected kwarg), so it’s not actually backward compatible.  

Could you adjust the design to avoid breaking subclasses (e.g., keep `_sample` signature stable or allow `**kwargs` / separate RNG helper), and also consider avoiding the new `sklearn` dependency/import in the sampling path (prefer NumPy RNG utilities if possible)?

**Comment by ghost** [comment]
Hi @fkiraly and @moksha-hub,

Since #722 has been open for a few weeks with the backward-compatibility concern from @arnavk23 unresolved, I'd like to propose a fix and submit a fresh PR if that's welcome.

The core issue: adding `random_state` to `_sample()` breaks all existing subclasses that define `_sample(self, n_samples=None)` — they raise `TypeError` on the unexpected kwarg.

My proposed approach keeps `_sample()` signature **completely unchanged**:
- Add `random_state=None` to the public `sample()` only
- Normalize it to `np.random.Generator` via a small helper (pure numpy, no sklearn dependency)
- Temporarily inject it as `self._sample_rng` via a context manager before calling `_sample()`
- Base `_sample()` reads `self._sample_rng` if set, falls back to `np.random` otherwise
- `Empirical._sample()` gets a one-line update to respect `self._sample_rng`

This means zero changes to any existing subclass signature. Happy to submit a PR with full tests if this design looks good to you.
